Lab1/Lab1/Models/WorkoutModel.m

```diff
@@ -224,8 +235,41 @@ - (void)populateWithSampleData {
         }
         [builtWorkouts addObject:wk];
     }
-    self.workouts = builtWorkouts;
+    
+    NSSortDescriptor *sortDescriptor;
+    sortDescriptor = [[NSSortDescriptor alloc] initWithKey:@"startTime"
+                                                 ascending:NO];
+    
+    self.workouts = [[builtWorkouts sortedArrayUsingDescriptors:@[sortDescriptor]] mutableCopy];
     NSLog(@"Generated workouts: %@", self.workouts);
 }
 
+
+// takes an array of WorkoutTableViewCells and saves them
+- (void)saveExercises:(NSMutableArray *)exercises withName:(NSString*)workoutName withStartDate:(NSDate*)startDate withEndDate:(NSDate*)endDate {
+    NSArray* names = [self getExerciseNames];
+
+    // create the workout
+    Workout *wk = [[Workout alloc] initWithEntity:self.workoutEntityDescription insertIntoManagedObjectContext:self.managedObjectContext];
+    wk.name = workoutName;
+    wk.startTime = startDate;
+    wk.endTime = endDate;
+    
+    // create the exercises
+    for (NewWorkoutTableViewCell *cell in exercises) {
+        Exercise *ex = [[Exercise alloc] initWithEntity:self.exerciseEntityDescription insertIntoManagedObjectContext:self.managedObjectContext];
+        ex.name = names[[cell.workoutTypePicker selectedRowInComponent:0]];
```

Comment:
This should check to see if an exercise with that name already exists. If I create a new exercise in the New Workout modal, and I set that exercise as a favorite, it is only reflected by that workout, and not in other workouts that have the same exercise.